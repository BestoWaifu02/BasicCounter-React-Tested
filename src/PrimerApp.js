import React from "react";
import PropTypes from "prop-types";
//Recibiendo PROPS, y props por default
const PrimerApp = ({ saludo, subtitle }) => {
  //const saludo = "Hola Mundo";

  return (
    /* Las llaves indican que se puede enviar cosas de JS en las TAGS
    Se pueden enviar  INTS,Strings,doubles pero NO Booleans    
    */
    <>
      <h1> {saludo} !!!</h1>
      {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
      <p>{subtitle}</p>
    </>
  );
};
//Al mandar props aqui definimos si son requeridas
PrimerApp.propTypes = {
  saludo: PropTypes.string.isRequired,
};
//En lugar de poner los props por default en los parametros se pueden definir con el nombre del componente.defaultProps
PrimerApp.defaultProps = {
  subtitle: "soy un subtitulo",
};

export default PrimerApp;
