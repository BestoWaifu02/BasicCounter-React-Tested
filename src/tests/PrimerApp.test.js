//import { render } from "@testing-library/react";
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { shallow } from "enzyme";
import PrimerApp from "../PrimerApp";

describe("Pruebas en el componente PrimerApp", () => {
  /* PRUEBA CON JEST */

  /*   test('Debe mostrar el mensaje "Hola soy Diego" ', () => {
    const saludo = "Hola soy Diego";
    //Render es una funcion que recibe un componente a renderizar
    //Mandamos las props del componente porque las definimos que requisito
    //Desestructuramos el getByText del render
    const { getByText } = render(<PrimerApp saludo="Hola soy Diego" />);
    expect(getByText(saludo)).toBeInTheDocument();
  }); */

  /* PRUEBA CON ENZYME */
  test("Debe mostrar primerApp correctamente", () => {
    const saludo = "Hola soy Diego";
    const wrapper = shallow(<PrimerApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el subtitulo enviado por PROPS", () => {
    const subtitle = "soy un subtitulo";
    const saludo = "Hola soy Diego";
    const wrapper = shallow(<PrimerApp saludo={saludo} subtitle={subtitle} />);
    //Aqui es como un doc.querySelector ya que busca en el wrapper
    const textoParrafo = wrapper.find("p").text();
    console.log(textoParrafo);
    expect(textoParrafo).toBe(subtitle);
  });
});
