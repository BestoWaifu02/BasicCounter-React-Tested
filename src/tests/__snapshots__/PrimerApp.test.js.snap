// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente PrimerApp Debe mostrar primerApp correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola soy Diego
     !!!
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
